### PR TITLE
main: add `make coverage_server` build option

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve? 
Add `coverage_server` option to Makefile

`make coverage_server` builds a special tidb-server binary which provides unit test argument options. For example, `-test.coverprofile=` can be used to generate raw coverage report after tidb-server shutdown. 

### What is changed and how it works?
A `tidb-server/main_test` is added, and it simply wraps the main entry of TiDB in a unit test function:

```
func TestRunMain(t *testing.T) {    
    main()    
}  
```

Then `make coverage_server` is added to Makefile and build `tidb-server/main_test` as a unit test binary.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
```
~/Projects/gopath/src/github.com/pingcap/tidb$ make coverage_server
```
The command above builds `bin/tidb-coverage-server`.

Code changes

N/A

Related changes

N/A

### CODE REVIEW

**Comment by bb7133** [comment]
PTAL @coocood

File: tidb-server/main.go [hunk 1]
```diff
@@ -178,15 +178,19 @@ func main() {
 	signal.SetupSignalHandler(serverShutdown)
 	runServer()
 	cleanup()
-	exit()
+	syncLog()
 }
 
 func exit() {
+	syncLog()
```
Comment (coocood) [inline tidb-server/main.go]: This change seems unrelated

Comment (bb7133) [reply tidb-server/main.go]: `exit()` is also called by `setCPUAffinity()`. This is to make sure the logic is the same for `setCPUAffinity()`

https://github.com/pingcap/tidb/blob/0ff5b50c1509bed279a2c2ba7acdb38519b38c9c/tidb-server/main.go#L192

**Comment by coocood** [comment]
LGTM

**Comment by zz-jason** [comment]
@bb7133 please add some proper labels.

**Comment by crazycs520** [comment]
@bb7133  Need cherry-pick 3.1?
